### PR TITLE
Prevents bees form synthesising forbidden reagents

### PR DESCRIPTION
**Pull request by Birdtalon**
A request.

Some reagents such as Adminordrazine are unable to be synthesised using an Odysseus for example with `can_synth = FALSE` it was brought to my attention that bees can be used to infinitely reproduce every reagent as they ignore this check.

- Prevents random spawn beeboxes from spawning with a forbidden reagent
- Prevents replicating bees with forbidden reagents using royal bee jelly.
- Queen bees wtih forbidden reagents placed into a hive will not take to it.
- If all else fails, if the hive has managed to get a queen with forbidden reagent (somehow) it won't produce honeycomb.

I think I've covered all eventualities here.

🆑 Birdtalon
fix: Bees can no longer reproduce un-synthable reagents.
/🆑

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/bees.dm [hunk 1]
```diff
@@ -253,6 +250,9 @@
 		if(S.reagents.has_reagent("royal_bee_jelly")) //checked twice, because I really don't want royal bee jelly to be duped
 			if(S.reagents.has_reagent("royal_bee_jelly",5))
 				S.reagents.remove_reagent("royal_bee_jelly", 5)
+				if(!queen.beegent.can_synth)
+					to_chat(user, "<span class='warning'>You inject [src] with the royal bee jelly. It's ineffecive! Maybe it's something to do with the [src] reagent.</span>")
```
Comment (FalseIncarnate) [inline code/modules/mob/living/simple_animal/hostile/bees.dm]: `ineffective`

Comment (Birdtalon) [reply code/modules/mob/living/simple_animal/hostile/bees.dm]: 👀

**Comment by mrturkeytoe** [comment]
remove funbab

**Comment by taukausanake** [comment]
I think it's fair for random boxes to not have those chems inside them but not so much for the chems to not be create-able by bees. This also sounds like it'll fuck up bees if you can't change their reagent from a forbidden one (not sure if you can either way, never played with that feature).
I understand the admin chem but what other examples are there? I can see admin bees beeing fun and this would stop it altogether. That's not fun

**Comment by Birdtalon** [comment]
Reagents with the `can_synth = FALSE` flag:

Adminordrazine
Dragon's Breath
All children of `/datum/reagent/blob`
spider eggs
Nanomachines
Xenomicrobes
Tubercle bacillus Cosmosis microbes
Jagged Crystals
Salmonella
Gibbis

**Comment by alexkar598** [comment]
That's all the fun ones!

**Comment by taukausanake** [comment]
Okay, not very fair for mutation and advanced mutation toxin because Xenobio. The only reason those are in the list is because they can be abused but then you can also have fun abuse. Hydro/Xenobio team up? RIP.
Nanomachines? But my borg bees. ☹️ 
Dragon's Breath? I didn't even know that was a reagent; I just thought it was a shotgun shell. I REALLY want to see dragon bees now!
Spider eggs? Spider bees!

I think I'm okay with just the first part of this PR, random bee boxes not having these chems. Maybe even replicating queen bees with these chems but only if it lets you make a vanilla bee instead of outright refusing. Removing all these fun possibilities is bad though, especially if it would otherwise be impossible or really difficult to get any of the chems to start with

**Comment by FalseIncarnate** [comment]
As is the case in the Silver Slime Core PR, the reason for these changes are because the vast majority of those listed chemicals are not meant to be obtainable by players, especially in large amounts. Blob reagents shouldn't exist outside of blobs and their victims, likewise an expensive traitor-only poison that comes in a limited quantity shouldn't be something you can produce infinitely through bee stings.

Plain and simple, its a fix for an exploit that had some serious balance ramifications if left unchecked. If you don't like it, chances are you were likely to be someone to abuse it.

**Comment by uraniummeltdown** [comment]
Nanomachine bees are a legit interesting hijack strat, at least let nanomachines be synthesised. They're easy to get anyway, not rare at all

**Comment by Birdtalon** [comment]
It's correct that discussions could be had about all those chems on the list. This PR ties up a loophole and is seperate from the discussion of whether or not those chems should be on the list or not.

File: code/modules/hydroponics/beekeeping/beebox.dm [hunk 2]
```diff
@@ -71,8 +71,12 @@
...
 	var/datum/reagent/R = null
 	if(random_reagent)
-		R = pick(subtypesof(/datum/reagent))
-		R = chemical_reagents_list[initial(R.id)]
+		var/synthable = FALSE
+		while(!synthable)
+			R = pick(subtypesof(/datum/reagent))
+			if(R.can_synth)
+				R = chemical_reagents_list[initial(R.id)]
+				synthable = TRUE
```
Comment (FalseIncarnate) [inline code/modules/hydroponics/beekeeping/beebox.dm]: You can condense this entire thing by just using `R = get_random_reagent_id()` which already checks against can_synth

**Comment by mrturkeytoe** [comment]
I see no issue needing fixing. This is a !!!FUN!!:tm: Feature:tm: meme

File: code/modules/hydroponics/beekeeping/beebox.dm (hunk 2 above)
Comment (Birdtalon) [reply code/modules/hydroponics/beekeeping/beebox.dm]: That's helpful!

**Comment by mrturkeytoe** [comment]
All I can say about this PR and the silver slime PR is this is the exact same argument made against gatfruit and history has proven gatfruit are not abused and this is 100% hugbox, bubblewrapping fun removal. Not to mention as someone on discord said, "They shouldn't be there because we didn't intend them to be there." Is basically a 'because I said so' line of thinking without a legitimate motivation.

**Comment by Spacemanspark** [comment]
Adminirdizine is literally on the same level as spawning in a pulse annihilator. It's an administrative only chemical, hence the name.
It was never meant to be synthesized.

**Comment by Anoonki** [comment]
@Spacemanspark Nukies have access to Adminordrazine.
I wonder how people keep forgetting that.

**Comment by tigercat2000** [comment]
@Anoonki How do you figure that?

**Comment by Anoonki** [comment]
@tigercat2000 Unless it was changed, their nukie medkit they can buy (Including splints, and an NV Medhud) gets a deathsquad nanite injector with a slightly lower capacity. 
Unless I'm mistaken, that's renamed Adminordrazine.

**Comment by Birdtalon** [comment]
I thought it was restorative nanites?

**Comment by Anoonki** [comment]
@Birdtalon While the code I have is a few months out of date, it shows up as just "Nanites"

**Comment by FalseIncarnate** [comment]
Syndicate nanites are different than normal nanites.

Syndicate nanites only deal damage, they do nothing for various side-effects or broken bones.

Nanites (the reskinned admin reagent) are only found in a special hypospray given to the highest level ERT (which requires admins to trigger/spawn), so they are effectively admin-only as well.

This may be your confusion.

**Comment by tigercat2000** [comment]
@Anoonki False is correct (sorry, missed your response)- They get an `
/obj/item/weapon/reagent_containers/hypospray/combat` injector, which just has the following:
`list_reagents = list("epinephrine" = 30, "omnizine" = 30, "teporone" = 15)`
They previously had synaptizine in them (over a year ago).

**Comment by FalseIncarnate** [comment]
@tigercat2000 @Anoonki I actually was referring to the Syndicate Nanites found in the medical syndiborg hypospray, not the hypospray in the op medkit. Though it pretty much is the same effect as omnizine, so we could probably swap those reagents with little to no ill effect (albeit in a different PR since that is entirely unrelated to bees).

**Comment by imsxz** [comment]
I dunno what all the fuss over this PR is. The chems that don't have can_synth or whatever aren't meant to be mass produced. The chems that can or can't be synthed should be discussed on the forum, as it is outside the scope of this PR.

**Comment by mrturkeytoe** [comment]
I mean, the discussion is relevant to the PR and I suspect if it's not discussed here it will be dismissed and never brought up ever again.

**Comment by tigercat2000** [comment]
This is an exploit. Individual reagent's can_synthesize properties can be discussed by themselves.

**Pull request merged**